test/linalg/diagonal.jl

```diff
@@ -158,6 +162,17 @@ srand(1)
         @test (r = full(D) * U   ; A_mul_B!(UU, D, U) ≈ r ≈ UU)
         @test (r = full(D)' * U  ; Ac_mul_B!(UU, D, U) ≈ r ≈ UU)
         @test (r = full(D).' * U ; At_mul_B!(UU, D, U) ≈ r ≈ UU)
+
+        VV = full(D)
+        DD = copy(D)
+        r  = VV * full(D)
+        @test full(A_mul_B!(VV, DD)) ≈ r ≈ full(D)*full(D)
+        DD = copy(D)
+        r  = VV * (full(D).')
+        @test full(A_mul_Bt!(VV, DD)) ≈ r
+        DD = copy(D)
+        r  = VV * (full(D)')
+        @test full(A_mul_Bc!(VV, DD)) ≈ r
```

Comment:
Perhaps disambiguate the `full` calls in this block? That is, perhaps use `Array(...)` or `AbstractArray(...)` as appropriate instead?